### PR TITLE
[#20] Implement pull to refresh surveys on Home screen

### PR DESCRIPTION
**Pull request by chornerman**
#20 

## What happened 👀

- [x] Add a refresh indicator for refreshing survey list on the Home screen using Lucus's workaround
- [x] Always jump to the first page when executing pull to refresh
- [x] When refreshing the survey list, refresh all loaded survey items
- [x] Pull to refresh functionality will only be enabled when the Home screen's state is `ApiLoadingSuccess` or `Error` (after finish loading survey list from the API)
- [x] Move a constant to a common `Constants` class
- [x] Add unit tests for `refreshSurveys` function

## Insight 📝

N/A

## Proof Of Work 📹

https://user-images.githubusercontent.com/13492460/205985059-333546c0-0ff3-417f-93db-8cd9f81eda6f.mp4

### CODE REVIEW

File: lib/api/repository/survey_repository.dart [hunk 1]
```diff
@@ -34,7 +33,9 @@ class SurveyRepositoryImpl extends SurveyRepository {
   }
 
   void _saveSurveysCache(int pageNumber, List<SurveyModel> surveyModels) {
```
Comment (markgravity) [inline lib/api/repository/survey_repository.dart:35]: What do you think about changing `int pageNumber` to `bool wantsClearCache`, passing a `pageNumber` here is quite redundant 🤔

**Review by markgravity — COMMENTED** [review]
The rest LGTM

**Comment by luongvo** [comment]
@chornerman pls rebase this PR for review

**Comment by chornerman** [comment]
> @chornerman pls rebase this PR for review

@luongvo I think I rebased it yesterday 🤔

**Comment by luongvo** [comment]
> > @chornerman pls rebase this PR for review
> 
> @luongvo I think I rebased it yesterday 🤔 Could you please help recheck 🙏🏼

My bad, I did not refresh the page 😛

**Review by markgravity — APPROVED** [review]

File: test/api/repository/survey_repository_test.dart [hunk 2]
```diff
@@ -24,7 +24,7 @@ void main() {
     });
 
     test(
-        'When calling getSurveys successfully, it returns corresponding mapped result',
+        'When calling getSurveys with pageNumber equal to 1 successfully, it returns corresponding mapped result and clear surveys cache',
```
Comment (luongvo) [inline test/api/repository/survey_repository_test.dart:27]: how about

```suggestion
        'When calling getSurveys on the first page successfully, it returns corresponding mapped result and clear surveys cache',
```

File: test/api/repository/survey_repository_test.dart [hunk 3]
```diff
@@ -41,6 +41,23 @@ void main() {
       verify(mockHiveUtils.saveSurveys(result)).called(1);
     });
 
+    test(
+        'When calling getSurveys with pageNumber not equal to 1 successfully, it returns corresponding mapped result',
```
Comment (luongvo) [inline test/api/repository/survey_repository_test.dart:45]:
```suggestion
        'When calling getSurveys not on the first page successfully, it returns corresponding mapped result',
```

**Review by luongvo — COMMENTED** [review]
rest lgtm

**Review by luongvo — APPROVED** [review]

**Pull request merged**